sigilrecorder.lic

```diff
@@ -0,0 +1,297 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#sigilrecorder
+=end
+
+custom_require.call(%w[common common-arcana common-items common-travel drinfomon events spellmonitor equipmanager])
+
+class SigilRecorder
+	require 'yaml'
+
+	def initialize
+    arg_definitions = [
+      [
+        { name: 'debug', regex: /debug/i, optional: true, description: 'Show debug messaging' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+    args.debug == 'debug' ? @debug = true : @debug = false
+
+		#initialize arrays for supported towns, sigils, and seasons
+		@city_list = ['Crossing', 'Riverhaven', 'Shard']
+		@sigil_list = ['abolition','congruence','induction','permutation','rarefaction', 'antipode','ascension','clarification','decay','evolution','integration','metamorphosis','nurture','paradox','unity']
+		@season_list = ['summer', 'fall', 'winter', 'spring']
+    @start_room = Room.current.id
+
+		#pull data and settings and store the current region and season
+		@data = get_data('sigils')
+		@settings = get_settings
+		@region = get_region #parse Room.current.image for which region/city we're in
+		@season = get_season #make sure we're recording sigils to the correct season
+    @hunting_zones = get_data('hunting').hunting_zones.values.flatten.uniq!
+		
+		#generate an array of all the room numbers associated with the current room's Room.current.image value
+		region_list = region_room_list
+		#subtract known hunting spots and any other rooms in the YAML's nogo list to avoid combat and rooms with known mapping issues
+		walk_list = generate_nogo_list(region_list)
+		
+		#debug info
+		DRC.message("Region detected: #{@region}") if @debug
+		DRC.message("Season detected: #{@season.capitalize}") if @debug
+
+		#perform any outdoorsmanship buffs as this is the skill that governs sigil hunting
+		DRC.message("Checking for 'outdoors' waggle set...") if @debug
+		DRCA.do_buffs(@settings, 'outdoors') #perc sigil uses Outdoorsmanship
+		
+		#flag known patterns that suggest the room is unsuitable for sigil hunting or harvesting
+		Flags.add('sigil-danger-room', /.*begins to advance on you.*|You weave back and forth in a dance-like motion|You must be closer to use tactical abilities on your opponent/)
+
+		#main travel loop, travels to rooms and then kicks off the sigil hunting process
+		walk_list.each do |roomid|
+			
+			#check if the room title is in the 'banpatterns' array before travelling there. Skip and add to nogo if a match is found
+			banflag = false
+			@data.SigilInfo['banpatterns'].each do |pattern|
+				if Map[roomid].title.to_s.include?(pattern)
+					prune_walk_list(roomid, @data)
+					DRC.message("Match found in 'banpatterns'. Adding #{roomid} to nogo.") if @debug
```

Comment:
💡 Based on the wording of the debug message, perhaps these two lines should be swapped?

Reply:
Updated. Also swapped the order of the sentences from a readability standpoint (i.e. put the "What I am doing"  before the "Why I am doing it.")